### PR TITLE
use label template helper from fmt & use labels & selectorLabels

### PR DESCRIPTION
**Pull request by ubergesundheit**
Use `_helpers.tpl` helm templates file from [`fmt`](https://github.com/giantswarm/fmt/blob/master/kubernetes/annotations_and_labels.md#labels-set-on-objects-installed-by-charts) and use the templates in chart yaml.

I'm still very new to this, @rossf7 could you take a look at this?

## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: patch/03.labels.patch [hunk 1]
```diff
@@ -0,0 +1,270 @@
+diff --git a/templates/_helpers.tpl b/templates/_helpers.tpl
```
Comment (rossf7) [inline patch/03.labels.patch:1]: Looks like a rogue file?

File: helm/gatekeeper-app/values.yaml [hunk 2]
```diff
@@ -17,3 +17,6 @@ resources:
   requests:
     cpu: 100m
     memory: 256Mi
+project:
+  branch: "[[ .Branch ]]"
+  commit: "[[ .SHA ]]"
```
Comment (rossf7) [inline helm/gatekeeper-app/values.yaml:22]: Good to add these but architect-orb is 0.6.0 which might miss support for these.
https://github.com/giantswarm/gatekeeper-app/blob/979eee960e0aa13de2d1f965ca46d57f51dcb990/.circleci/config.yml#L3

Can you bump it to 0.8.9 which is the latest?

File: patch/03.labels.patch (hunk 1 above)
Comment (MarcelMue) [inline patch/03.labels.patch:1]: Why is this needed?

Comment (ubergesundheit) [reply patch/03.labels.patch:1]: The original creator of this repository tried to sync up this chart with the upstream chart of gatekeeper. The script used for this is [here]https://github.com/giantswarm/gatekeeper-app/blob/master/script/sync_chart.sh). There are [other patches in the directory](https://github.com/giantswarm/gatekeeper-app/tree/master/patch)

File: helm/gatekeeper-app/values.yaml (hunk 2 above)
Comment (ubergesundheit) [reply helm/gatekeeper-app/values.yaml:22]: yes

**Comment by ubergesundheit** [comment]
Fingers crossed, this should work as is. One part of the CI process executes `helm template`. No errors here (fixed in https://github.com/giantswarm/gatekeeper-app/pull/30/commits/c311dac641da8cef15efb2fa7b8f42773f6e2c48)

I'm still undecided if I should bump the version in here or do a `release-v0.3.0` branch after merging this PR in to master
